src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp

```diff
@@ -125,6 +125,12 @@ mp::QemuPlatformDetail::QemuPlatformDetail(const mp::Path& data_dir)
 
 mp::QemuPlatformDetail::~QemuPlatformDetail()
 {
+    for (const auto& it : name_to_net_device_map)
+    {
+        const auto& [tap_device_name, hw_addr] = it.second;
+        remove_tap_device(tap_device_name);
+    }
```

Comment:
This we should've done unrelated to the bug, IIUC?

Reply:
No, because this is the whole reason why I called `remove_resources_for()` in the `QemuVirtualMachine` dtor to begin with.  Now that there is a bug actually calling that, the logic needs to be moved here.